### PR TITLE
chore: Add isSystemGenerated field to analytics for git operations

### LINKED ISSUE #13477: [Task]: Add `isSystemGenerated` field in git-sync analytics events
## Description
We want to distinguish the events generated from Appsmith vs the user-generated events to optimize the mixpanel funnels.

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description

> We want to distinguish the events generated from Appsmith vs the user-generated events to optimize the mixpanel funnels.

Fixes #13477 

## Type of change

> Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java [hunk 1]
```diff
@@ -503,12 +509,14 @@ public Mono<String> commitApplication(GitCommitDTO commitDTO, String defaultAppl
...
                                 "up a git profile from the user profile section.";
+
                         return addAnalyticsForGitOperation(
                                 AnalyticsEvents.GIT_COMMIT.getEventName(),
                                 childApplication,
                                 AppsmithError.INVALID_GIT_CONFIGURATION.getErrorType(),
                                 AppsmithError.INVALID_GIT_CONFIGURATION.getMessage(errorMessage),
-                                childApplication.getGitApplicationMetadata().getIsRepoPrivate()
+                                childApplication.getGitApplicationMetadata().getIsRepoPrivate(),
+                                isSystemGenerated
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:519]: Why are we adding the `isSystemGenerated` for error events?

**Review by abhvsn — CHANGES_REQUESTED** [review]
As we are using the `isSystemGenerated` flag only in commit flow, does it make sense to keep 2 methods:
1. addAnalyticsForGitOperation(String eventName, Application application, Boolean isRepoPrivate, Boolean isSystemGenerated)
2. addAnalyticsForGitOperation(String eventName, Application application, Boolean isRepoPrivate)

This will prevent us from making larger change if this is only specific to certain flow

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java (hunk 1 above)
Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/GitServiceCEImpl.java:519]: It is set false. But yeah i get what you are saying.

**Comment by AnaghHegde** [comment]
@abhvsn Only commit is system generated action. All other cases are user generated. I would add a new method just for commit operation.

**Comment by abhvsn** [comment]
@AnaghHegde can you please resolve the conflicts, I can start the review once this is done

**Comment by AnaghHegde** [comment]
@abhvsn I have not made any changes. I want you to reply for the comments.

**Comment by abhvsn** [comment]
> @abhvsn Only commit is system generated action. All other cases are user generated. I could add a new method just for commit operation. But if i do that then `isSystemGenerated` will be present for few cases and not at all present in other cases. What if i add one method where we set this only for commit operation. This way we would not have to deal with the problem of having to check if the value is present or not while building the charts in mixpanel.

Is this different from what I was expecting in above comment? Can you please elaborate how your approach will be different than what I was expecting here. The other part where you were asking for having `isSystemGenerated` boolean for all the events still can be achieved by specifically setting that variable in 
`addAnalyticsForGitOperation(String eventName, Application application, Boolean isRepoPrivate)`
https://github.com/appsmithorg/appsmith/pull/13705#pullrequestreview-970319295

**Comment by AnaghHegde** [comment]
@abhvsn per your suggestion we would still be passing the default values to the method where we actually call the analytics service method. I am suggesting we would have a separate implementation just for the commit flow.

**Review by abhvsn — APPROVED** [review]

**Pull request merged**